internal/runner/rspec.go

```diff
@@ -89,17 +92,28 @@ func (r Rspec) Run(testCases []string, retry bool) (RunResult, error) {
 		return RunResult{Status: RunStatusError}, fmt.Errorf("failed to build command: %w", err)
 	}
 
-	// Create a temporary file to store the JSON output of the rspec run.
-	// This is a temporary solution, until we have an option to specify the output file.
-	f, err := os.CreateTemp("", "rspec-*.json")
-	if err != nil {
-		return RunResult{Status: RunStatusError}, fmt.Errorf("failed to create temporary file for rspec output: %v", err)
+	var resultPath string
+
+	if r.ResultPath != "" {
```

Comment:
Should this whole thing be wrapped in an `if retry`? Like we only have to write the file if the customer wants retries, right?

Reply:
Good point. Currently, TSC will write the output even when retry is disabled, as I think it will be useful to save the output regardless. Do you think we should save the output only when the retry is enabled for now?

Reply:
I think it would possibly be a bit surprising if we created a file and didn't tell the user about it. I think I'd maybe design it like if the customer specifies a file name, then the file is saved, but if they don't, the file is deleted? I wouldn't want a side effect of using the TSC be that we clutter a host machine with lots of files that the user didn't know we were creating? What do you think?

Reply:
I think we should save the output if the user manually specifies the file path for the output rather than if retry is enabled.

Reply:
Yup that sounds good!

Reply:
Yeah, that's a good point, I considered this option when I implemented these changes. Currently, it's redundant when retry is disabled and no file path is specified. However, this will be useful in the future when we want to do something with the output, for example to send the output to Test Analytics. Although it's more like a future problem, I'd like to keep it this way for the simplicity.

> Unless there's some unintended side effects of creating and then deleting this temp file? Like permissions or in the unlikely event that file creation/deletion crashes?

There is always a risk, but I'd say the risk is low. We are using go std library to create/delete the temp file, which I believe is carefully implemented and tested.

Reply:
Awesome to hear you've though through all the trade-offs and future possibilities 👍